master/internal/configpolicy/utils.go

```diff
@@ -180,25 +242,16 @@ func checkConstraintConflicts(constraints *model.Constraints, maxSlots, slots, p
 	return nil
 }
 
-// checkAgainstGlobalConfig is a generic to check constraints & invariant configs against the global config.
-func checkAgainstGlobalConfig[T any](
-	globalConfigPolicies *string,
-	config *T,
-	errorMsg string,
-) {
-	if globalConfigPolicies != nil && config != nil {
-		global, err := UnmarshalConfigPolicy[T](*globalConfigPolicies, errorMsg)
-		if err != nil {
-			ConfigPolicyWarning(err.Error())
-			return
-		}
-		configPolicyConflict(global, config)
+// warnConfigPolicyOverlap compares two different configurations and
+// warns the user when both configurations define the same field.
+func warnConfigPolicyOverlap(config1, config2 interface{}) {
```

Comment:
I prefer `configPolicyConflict` over `warnConfigPolicyOverlap`. Simply `configPolicyOverlap` works too. My reasoning is that if we change the logic and decide to error or something, we now have to change the function name ... even though the function is not meaningfully different.

Reply:
Makes sense! I think the function's purpose was easier to understand with the word "overlap", and added warn so that the function was a verb, rather than a noun. 
I'll change the name to `configPoliciyOverlap`!